packages/gatsby-plugin-manifest/src/gatsby-node.js

```diff
@@ -43,8 +43,22 @@ function generateIcons(icons, srcIcon) {
   )
 }
 
-exports.onPostBootstrap = async ({ reporter }, pluginOptions) => {
-  const { icon, ...manifest } = pluginOptions
+exports.onPostBootstrap = async ({ reporter }, { manifests, ...manifest }) => {
+  const activity = reporter.activityTimer(`Build manifest and related icons`)
+  activity.start()
+
+  if (Array.isArray(manifests)) {
+    await Promise.all(manifests.map(x => makeManifest(reporter, x)))
```

Comment:
So currently this would run the icon creation for every manifest. The icons aren't being internationalized. In automatic mode they couldn't be without code modification. I do see a use case for i18n of icons...but for those who aren't creating all these icons over and over seems pointless. We should probably add a caching mechanism to fix this. That can be a separate PR though. no need to solve it here as it technically affects the current plugin regenerating icons on every build currently even if they already exist.

For now, should we be concerned with adding the i18n suffix to icons? This would fix the automatic mode issue of writing all icons to the same files.

Reply:
Haven't thought about the icon suffix 😨 Can do this tomorrow 👍 I didn't even include the icon in the i18n readme example, could be more complex than maybe necessary, but we could check where the icon has been defined, as shared or language specific and add the suffix only in the latter case.. what do you think? 🤔

Reply:
That's a great idea. Suggestion: process the "default" first (i.e. what's in the options and not the "manifest" array). Then as you iterate through the `manifest: []` only if icon is uniquely specified in there or if `icons` is specified do you generate new icons...otherwise you just reference the default ones.

Does that make sense? there maybe other things to consider in this implementation. It would also be simpler to just always create unique icons. :man_shrugging:

Reply:
Realized the icon suffix is still missing..so sorry for the ping^^ 🙏

Reply:
Hum, any thoughts on it? Right now as we're merging the defaults directly with the localized versions I'd have to probably change the way data gets passed to [makeManifest](https://github.com/gatsbyjs/gatsby/pull/13471/files#diff-4606cc9066942a1129552adc6bd0b87dR54)

> It would also be simpler to just always create unique icons.

Any news on how do you feel about it?

Reply:
I think providing unique icons for a language is some what rare. Additionally, creating unique icons by default sounds generally wasteful, especially if no one is ever needing this. 

Basically, no matter how we do this we have to track what's been done already. We could do it like I suggested above...or we could check the input file and see if its digest has already been seen and written to disk, if so...then we use default file paths, otherwise we generate new file paths. Then add them the new digest to running log. If the digest is used again we;ll know when can use that same path.

Reply:
@CanRau Has this been resolved?

Reply:
No, you're right, only thing is that you're now able to define a default icon for all languages, which would still run on every language

So, I'm going to implement

> process the "default" first (i.e. what's in the options and not the "manifest" array). Then as you iterate through the manifest: [] only if icon is uniquely specified in there or if icons is specified do you generate new icons...otherwise you just reference the default ones

then?

Reply:
Dude, this is so simple and clean I'm slightly embarrassed I didn't think of it...good work. Once this is done I'll do a final review, but atm I think that covers all the issues.

Reply:
Not sure I get your comment but I just quoted your suggestion https://github.com/gatsbyjs/gatsby/pull/13471#discussion_r277966669 😉

Reply:
Okay, this should be resolved. I ended up just requiring the name based cache busting on non-default locales with unique icons (automatic mode only) to keep icons from being overridden.